rust/src/treefile.rs

```diff
@@ -2931,6 +2936,17 @@ pub(crate) mod tests {
         }
     }
 
+    #[test]
+    fn ostree_layer_hashes() {
```

Comment:
One thing that would help build confidence here is some new unit tests (in a separate PR) that validate the checksum *does* change when e.g. a postprocess script changes, etc.

Reply:
Ok will followup with a new PR